### ISSUE TITLE
New rule jsx-ban-props prohibit specific prop names

### ISSUE THREAD

**Issue by gruckion** [yannickcr/eslint-plugin-react#2306]
Similar to the [tslint react](https://github.com/palantir/tslint-react) propose [forbid-prop-types](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-prop-types.md). Allows prohibiting of props in JSX with an optional explanatory message in the reported failure.

Example: block the use of `styles`, we do not want to allow the use of inline styles. There are a few edge cases where this might be suitable and we will disable the rule in those specific edge cases.

Configuration:

```json
{
  "jsx-ban-props": [true, ["style", "use className and css"]],
}
```

Error

```typescript
return (
  <div style={{ color: "white }}>
    
  </div>
)
```

Allowed

```typescript
return (
  <div className="colorWhite">
    
  </div>
)
```

**Comment by golopot** [comment]
Maybe this rule: [forbid-dom-props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-dom-props.md). It does not allow you to customize the error message though.

**Comment by ljharb** [comment]
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-component-props.md already exists.
